src/components/hand-poses.js

```diff
@@ -0,0 +1,125 @@
+const POSES = {
+  open: "allOpen",
+  thumbDown: "thumbDown",
+  indexDown: "indexDown",
+  mrpDown: "mrpDown",
+  thumbUp: "thumbsUp",
+  point: "point",
+  fist: "allGrip",
+  pinch: "pinch"
+};
+
+AFRAME.registerComponent("hand-poses", {
+  schema: {
+    leftPose: { type: "string", default: "allOpen" },
+    rightPose: { type: "string", default: "allOpen" },
+    mixer: { type: "string" },
+    gltfEntity: { type: "string", default: "a-gltf-entity" }
+  },
+
+  init() {
+    this.animatePose = this.animatePose.bind(this);
+    this.animatePoses = this.animatePoses.bind(this);
+    this.firstUpdate = this.firstUpdate.bind(this);
+    const mixerEl = this.el.querySelector(this.data.mixer);
+    if (mixerEl) {
+      this.mixer = mixerEl.components["animation-mixer"];
+    }
+
+    const onLoad = () => {
+      const mixerEl = this.el.querySelector(this.data.mixer);
+      if (mixerEl) {
+        this.mixer = mixerEl.components["animation-mixer"];
+      }
+      if (!this.mixer) {
+        // Can't find the mixer until spawned into the scene.
+        // TODO: Figure out the right event to listen to to get this done.
+        window.setTimeout(onLoad, 100);
+        return;
+      }
+      this.el.querySelector(this.data.gltfEntity).removeEventListener("loaded", onLoad);
+    };
+    this.el.querySelector(this.data.gltfEntity).addEventListener("loaded", onLoad);
+
+    //TODO: the loaded event isn't being caught, so we do it in a timeout
+    window.setTimeout(onLoad, 10);
+  },
+
+  update(oldData) {
+    if (!this.mixer) return;
+    if (!this.clipActionObject) {
+      const object3DMap = this.mixer.el.object3DMap;
+      const rootObj = object3DMap.mesh || object3DMap.scene;
+      this.clipActionObject = rootObj.parent;
+      this.firstUpdate();
+    } else {
+      this.animatePoses(oldData);
+    }
+  },
+
+  firstUpdate() {
+    this.leftClipFrom = this.leftClipTo = this.mixer.mixer.clipAction(POSES.open + "_L", this.clipActionObject);
```

Comment:
The "from" clips don't appear to ever be used, and the 'to" clips seems like they could be just local variables.

Reply:
Nevermind, found the dynamic access to them.. but that feels not ideal